control_unit.py

```diff
@@ -67,22 +67,33 @@ def _STR(instruction):
 
 
 def _RTI(instruction):
-    """unused"""
-    pass  # TODO throw exception here.
+    raise Exception("instruction is not implemented !")
+    pass  
 
 
 def _NOT(instruction):
-    """bitwise not"""
+    DR = (_NOT(instruction) >> 9) & 0X7                            # Destenation Register.
+    SR = (_NOT(instruction) >> 6) & 0X7                            # Source Register (register cntaining the data).
+    for B in range (16):                                           # B stands (in this line) for Bit.
+        reg_read(Registers(DR))[B] = ~reg_read(Registers(SR))[B]   # make every B in DR equals to the flip of the B in the same index in SR.   
```

Comment:
btw, `Registers(DR)` is not a list, You can't access it with `[]`.

Reply:
isn't every register an array itself ?

Reply:
Nope, every register is an `enum` with a constant value that represents it's index in the internal registers array.